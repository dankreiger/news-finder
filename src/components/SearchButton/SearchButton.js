import React from 'react';
import { func, string } from 'prop-types';
import { Button } from 'reactstrap';

const SearchButton = ({ handleClick, text, theme }) => (
  <Button color={theme} onClick={handleClick}>
    {text}
  </Button>
);

SearchButton.propTypes = {
  handleClick: func.isRequired,
  text: string,
  theme: string
};

SearchButton.defaultProps = {
  theme: 'primary',
  text: 'Click Me'
};

export default SearchButton;
