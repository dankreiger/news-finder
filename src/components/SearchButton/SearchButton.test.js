import React from 'react';
import SearchButton from './SearchButton';
import { Button } from 'reactstrap';
import { shallow } from 'enzyme';

describe('SearchButton', () => {
  let component;
  let button;
  const fetchArticles = jest.fn();
  const handleClick = jest.fn();

  beforeEach(() => {
    component = shallow(<SearchButton handleClick={handleClick} />);
    button = component.find(Button);
  });

  it('renders without crashing', () => {
    expect(component.length).toBe(1);
    expect(component).toMatchSnapshot();
  });

  it('renders an Button component', () => {
    expect(button.length).toBe(1);
  });

  it('calls handleClick function when Button is clicked', () => {
    button.simulate('click');
    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
