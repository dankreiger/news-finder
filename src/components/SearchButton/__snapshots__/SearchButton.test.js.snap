// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchButton
    handleClick={[MockFunction]}
    text="Click Me"
    theme="primary"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Click Me",
      "color": "primary",
      "onClick": [MockFunction],
      "tag": "button",
    },
    "ref": null,
    "rendered": "Click Me",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Click Me",
        "color": "primary",
        "onClick": [MockFunction],
        "tag": "button",
      },
      "ref": null,
      "rendered": "Click Me",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
